app/services/host_counter.rb

```diff
@@ -0,0 +1,15 @@
+class HostCounter
+  def initialize(association)
+    @association = association
+  end
+
+  def number_of_hosts(association)
+    counted_hosts[association.id] || 0
+  end
+
+  private
+
+  def counted_hosts
+    @counted_hosts ||= Host::Managed.group("#{Host.table_name}.#{@association}_id").count
```

Comment:
based on @tbrisker comment, I wonder if this needs to take into account permission checking, similar to https://github.com/theforeman/foreman/blob/develop/app/helpers/application_helper.rb#L493